core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -518,4 +497,49 @@ private void checkUserFileBufferBytes() {
     Preconditions.checkArgument((usrFileBufferBytes & Integer.MAX_VALUE) == usrFileBufferBytes,
         "Invalid \"" + Constants.USER_FILE_BUFFER_BYTES + "\": " + usrFileBufferBytes);
   }
+
+  /**
+   * Factory class to create configuration instances.
+   */
+  @ThreadSafe
+  public static final class Factory {
+
+    /**
+     * @param properties the source configuration
+     * @return a new configuration with only properties copied from source
+     */
+    public static Configuration create(Map<?, ?> properties) {
+      return new Configuration(properties);
+    }
+
+    /**
+     * @param conf the source configuration
+     * @return a new configuration with only properties copied from source
+     */
+    public static Configuration create(Configuration conf) {
```

Comment:
after doing some search, I think implementing `clone` is not considered  a good practice compared to using copy factory method or constructor. more details can be found 
http://stackoverflow.com/questions/11905630/java-super-clone-method-and-inheritance